dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Current/PhaseShifterBlockingI.mo

```diff
@@ -0,0 +1,47 @@
+within Dynawo.Electrical.Controls.Current;
+
+/*
+* Copyright (c) 2015-2021, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+model PhaseShifterBlockingI "Phase Shifter blocking model"
+  import Modelica.Constants;
+  import Dynawo.NonElectrical.Logs.Timeline;
+  import Dynawo.NonElectrical.Logs.TimelineKeys;
+  import Dynawo.Types;
+
+  public
+    parameter Types.CurrentModule IMax "Maximum current on the monitored component";
+    parameter Types.Time tLagBeforeActing "Time lag before taking action";
+
+    input Types.CurrentModule IMonitored "Monitored current";
+    output Boolean locked "Is phase shifter locked?";
+
+  protected
+    discrete Types.Time tThresholdReached (start = Constants.inf) "Time when I > IMax was first reached";
```

Comment:
I think the "discrete" keyword is there in the tap changer and phase shifter other implementations if I remember well. Nevertheless, I don't know what is its effect and I'm not sure it is necessary to have it. I would be in favor of deleting it. @cossartque do you know what it does?

Reply:
We can also keep the discrete here for the moment to be consistent with other parts.